src/main/java/lotto/type/GenerateType.java

```diff
@@ -0,0 +1,37 @@
+package lotto.type;
+
+import lotto.model.BuyingPocket;
+import lotto.model.Lottos;
+import lotto.service.LottosGenerator;
+import lotto.service.impl.LottosBothGeneratorImpl;
+import lotto.service.impl.LottosPassiveGeneratorImpl;
+import lotto.service.impl.LottosRandomGeneratorImpl;
+
+public enum GenerateType {
+    RANDOM(new LottosRandomGeneratorImpl()){
+        @Override
+        public Lottos generateLottos(BuyingPocket pocket) {
+            return lottosGenerator.generate(pocket);
+        }
+    },
+    PASSIVE(new LottosPassiveGeneratorImpl()){
+        @Override
+        public Lottos generateLottos(BuyingPocket pocket) {
+            return lottosGenerator.generate(pocket);
+        }
+    },
+    BOTH(new LottosBothGeneratorImpl()){
+        @Override
+        public Lottos generateLottos(BuyingPocket pocket) {
+            return lottosGenerator.generate(pocket);
+        }
+    };
+
+    LottosGenerator lottosGenerator;
+
+    GenerateType(final LottosGenerator lottosGenerator){
+        this.lottosGenerator = lottosGenerator;
+    }
+
+    public abstract Lottos generateLottos(BuyingPocket pocket);
```

Comment:
추상메소드로 선언해서 각각 메소드를 구현할 필요없이 이렇게 쓰면 되지 않을까요?
```suggestion
    public Lottos generateLottos(BuyingPocket pocket) {
    return lottosGenerator.generate(pocket);
    }
```

Reply:
음, 지금 구조는 `lottosGenerator`  클래스를 추상화함이, 크게 의미가 없지 않을까 생각합니당..

오브젝트 내용을 빌리면,
> 상속은 절차 추상화(메소드 추상화)를 통해서 다형성을 통해, `새로운 구현체의 확장`에 유리하지만,
> 데이터 추상화(멤버변수로 `enum` 과 같읕 타입을 정의해) `새로운 기능에 확장`에 유리합니다.

지금은, 데이터추상화에, 절차 추상화가 겹쳐있어, 어느쪽에도 유리하지 않은 구조가 아닐까합니다.ㅠ

결국 새로운 타입이는다면,
`Lottos`에서 데이터 추상화인 
새로운 분기 메소드가 필요합니다. (`createByPassive`, `createByBoth`, `createByRandom`),
더해서 절차 추상화인 `Generator`로 구현체도 필요하게됩니다

추후에도 로또발급에는 세가지 경우만 있다 가정하여,
 `데이터추상화`를 선택하게된다면, 
다음과 같이, `Generator`가 불필요한 구조로 변경 할수도있을것 같습니다. 
```
public Lottos generateLottos(BuyingPocket pocket) {
          return Lottos.create(this, pocket);
}

public class Lotto{
  public static create(GenerateType type, BuyPocket pocket){
    swtich(type){
    }
  }
}
```